test/library/misc.lua

```diff
@@ -0,0 +1,24 @@
+-- tests misc functions added by dfhack.lua
```

Comment:
I don't see any tests for a non-`pairs` iterator_fn, so maybe one could be useful.

If we want to test for the possible bug I identified above, I think we would need a type where pairs() errors and ipairs() is supported. I'm not aware of an existing one, but something like this could work:

```lua
local t = {1, 2}
setmetatable(t, {__pairs = function() error() end})
```

Reply:
Here's a more complete test I came up with, and I also confirmed that the above is a bug:
```lua
function test.safe_pairs_ipairs()
    local t = {1, 2}
    setmetatable(t, {
        __pairs = function()
            expect.fail('pairs() should not be called')
        end,
    })

    local iterated = 0
    for k,v in safe_pairs(t, ipairs) do
        expect.eq(k, v)
        iterated = iterated + 1
    end
    expect.eq(#t, iterated)
end
```